x/erc20/keeper/ibc_callbacks_test.go

```diff
@@ -2,574 +2,482 @@ package keeper_test
+			transferBytes: ibcgotesting.MockPacketData,
+			expectedError: true,
 		},
+		// Package has an invalid sender
 		{
-			name: "error - invalid sender (bad address)",
-			malleate: func() {
-				transfer := transfertypes.NewFungibleTokenPacketData(registeredDenom, "100", "badba1sv9m0g7ycejwr3s369km58h5qe7xj77hvcxrms", ethsecpAddrCosmos, "")
-				bz := transfertypes.ModuleCdc.MustMarshalJSON(&transfer)
-				packet = channeltypes.NewPacket(bz, 100, transfertypes.PortID, sourceChannel, transfertypes.PortID, evmosChannel, timeoutHeight, 0)
-			},
-			receiver:      secpAddr,
-			ackSuccess:    false,
-			checkBalances: false,
-			expErc20s:     big.NewInt(0),
-			expCoins:      coins,
+			name: "error - invalid sender",
+			transferBytes: newTransferBytes(
+				transfertypes.NewFungibleTokenPacketData(
+					unitNetwork.GetDenom(),
+					amountValue,
+					"",
+					keyring.GetAccAddr(1).String(),
+					"",
+				),
+			),
+			expectedError: true,
 		},
+		// Package has an invalid receiver
 		{
-			name: "error - invalid recipient (bad address)",
-			malleate: func() {
-				transfer := transfertypes.NewFungibleTokenPacketData(registeredDenom, "100", ethsecpAddrEvmos, "badbadhf0468jjpe6m6vx38s97z2qqe8ldu0njdyf625", "")
-				bz := transfertypes.ModuleCdc.MustMarshalJSON(&transfer)
-				packet = channeltypes.NewPacket(bz, 100, transfertypes.PortID, sourceChannel, transfertypes.PortID, evmosChannel, timeoutHeight, 0)
-			},
-			receiver:      secpAddr,
-			ackSuccess:    false,
-			checkBalances: false,
-			expErc20s:     big.NewInt(0),
-			expCoins:      coins,
+			name: "error - invalid receiver",
+			transferBytes: newTransferBytes(
+				transfertypes.NewFungibleTokenPacketData(
+					unitNetwork.GetDenom(),
+					amountValue,
+					keyring.GetAccAddr(0).String(),
+					"",
+					"",
+				),
+			),
+			expectedError: true,
 		},
+		// If we received an IBC from non EVM channel the account should be different
+		// If its the same, users can have their funds stuck since they dont have access
+		// to the same priv key
 		{
-			name: "no-op - sender == receiver, not from Evm channel",
-			malleate: func() {
-				transfer := transfertypes.NewFungibleTokenPacketData(registeredDenom, "100", ethsecpAddrEvmos, ethsecpAddrCosmos, "")
-				bz := transfertypes.ModuleCdc.MustMarshalJSON(&transfer)
-				packet = channeltypes.NewPacket(bz, 1, transfertypes.PortID, sourceChannel, transfertypes.PortID, "channel-100", timeoutHeight, 0)
-			},
-			ackSuccess:    true,
-			receiver:      secpAddr,
-			expErc20s:     big.NewInt(0),
-			expCoins:      coins,
-			checkBalances: true,
+			name: "error - sender == receiver from non EVM channel",
+			transferBytes: newTransferBytes(
+				transfertypes.NewFungibleTokenPacketData(
+					unitNetwork.GetDenom(),
+					amountValue,
+					keyring.GetAccAddr(0).String(),
+					keyring.GetAccAddr(0).String(),
+					"",
+				),
+			),
+			expectedError: false,
 		},
+		// TODO: Check why we do this
```

Comment:
What is the status on this TODO?

Reply:
The check `sender is module account` is a safeguards. I dont think we actually need the check, since currently there is no case of a module sending an IBC transaction. 
We can keep it in case of any future implementation or delete it

Reply:
ahh okay, yeah in that case I would prefer to keep it included for future purposes 👍